api/src/db/serde.rs

```diff
@@ -0,0 +1,93 @@
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use thiserror::Error;
+
+use crate::encryption::{
+    Decryptable, DecryptionError, Encryptable, EncryptionError, EncryptionKey,
+};
+
+/// Errors that can occur during serialization or encryption for database storage.
+#[derive(Debug, Error)]
+pub enum DbSerializationError {
+    /// Error occurred while serializing data to JSON for the database.
+    #[error("Error while serializing data to the db: {0}")]
+    Serde(#[from] serde_json::Error),
+
+    /// Error occurred while encrypting data for the database representation.
+    #[error("An error occurred while encrypting data for the db representation: {0}")]
+    Encryption(#[from] EncryptionError),
+}
+
+/// Errors that can occur during deserialization or decryption from database values.
+#[derive(Debug, Error)]
+pub enum DbDeserializationError {
+    /// Error occurred while deserializing data from JSON from the database.
+    #[error("Error while deserializing data from the db: {0}")]
+    Serde(#[from] serde_json::Error),
+
+    /// Error occurred while decrypting data from the database representation.
+    #[error("An error occurred while decrypting data from the db representation: {0}")]
+    Decryption(#[from] DecryptionError),
+}
+
+/// Serializes a value to a [`serde_json::Value`] for database storage.
+///
+/// Returns an error if serialization fails.
+pub fn serialize<S>(value: S) -> Result<serde_json::Value, DbSerializationError>
```

Comment:
This was created as a wrapper with the idea that if we want to add common serde logic also for non-encrypted structs we can easily modify this method and all usages will have the capability immediately.